### PR TITLE
Added Diagnosis at Discharge field when patient is discharged

### LINKED ISSUE #8299: Discharge diagnosis
**Is your feature request related to a problem? Please describe.**
When a patient is admitted, the diagnosis is marked by the doctor usually at the time of first consultation. This is then revised multiple times through the admission. Both the diagnosis itself or the diagnosis type alone may be updated during the term of admission.

It is important to prompt the user to update the diagnosis for a final time right before discharge. 

This is important since the patient may not be completely disease-free at discharge. He may simply be free of the life-threatening condition but may still have many underlying chronic illnesses for which the doctor may have suggested some medication in the discharge prescription field. 

**Describe the solution you'd like**
For an encounter with "Admission", based on the discharge reason, make the following edits in the discharge pop-up

1. *RECOVERED*

- [ ] Add a section for 'Diagnosis at discharge'- show the latest list of diagnoses (along with type) and allow user to make changes
- [ ] reorder the form in the following way
- Reason
- date and time of discharge
- Diagnosis at discharge
- Discharge prescription
- Discharge advice

2. *REFERRAL*

- [ ] Add 'Diagnosis at discharge' below date and time of discharge - show the latest list of diagnoses (along with type) and allow user to make changes

3. *LAMA*

- [ ] Add 'Diagnosis at discharge' below date and time of discharge - show the latest list of diagnoses (along with type) and allow user to make changes

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by JOSHIK27**
## Proposed Changes

- Fixes #8299 

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Facility/DischargeModal.tsx [hunk 1]
```diff
@@ -95,6 +99,15 @@ const DischargeModal = ({
     setFacility(referred_to);
   }, [referred_to]);
 
+  const { data } = useQuery(routes.getConsultation, {
+    pathParams: {
+      id: consultationData.id,
+    },
+  });
+
+  const ConsultationDiagnosisList: ConsultationDiagnosis[] =
```
Comment (rithviknishad) [inline src/Components/Facility/DischargeModal.tsx]: Ensure proper naming conventions for variables. And rely on type inference as useQuery is typesafe.

Ref: https://www.w3schools.com/js/js_conventions.asp#:~:text=Variable%20and%20function%20names%20written%20as%20camelCase

```suggestion
  const diagnoses =
```

File: src/Components/Facility/DischargeModal.tsx [hunk 2]
```diff
@@ -95,6 +99,15 @@ const DischargeModal = ({
...
+    data?.diagnoses?.map((diagnosis) => diagnosis) || [];
```
Comment (rithviknishad) [inline src/Components/Facility/DischargeModal.tsx]: Why is there an unnecessary `.map` call?
```suggestion
    data?.diagnoses ?? [];
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Facility/DischargeModal.tsx (hunk 2 above)
Comment (JOSHIK27) [reply src/Components/Facility/DischargeModal.tsx]: My bad, i was return something from each diagnosis item to populate into new array. Forget to change it later. Commited as suggested now. Thanks

**Review by rithviknishad — APPROVED** [review]

**Review by shivankacker — APPROVED** [review]

**Comment by samholics** [comment]
can we rename the "Diagnoses" to "Diagnosis at Discharge" pls? 
image

**Comment by rithviknishad** [comment]
> can we rename the "Diagnoses" to "Diagnosis at Discharge" pls?

@samholics Done

**Comment by samholics** [comment]
Tested working fine @rithviknishad @nihal467

**Comment by nihal467** [comment]
@JOSHIK27 

image

- [ ] When we open an active consultation page, it is throwing not found an error with 404
- [ ] the cypress was failing as well, verify the test is also passing

CC: @rithviknishad

**Comment by nihal467** [comment]
@JOSHIK27 are you still working on this issue ?

**Comment by JOSHIK27** [comment]
> @JOSHIK27 are you still working on this issue ?

Yes. I am

**Comment by nihal467** [comment]
@JOSHIK27 once the PR is ready for review, drop a comment

**Comment by JOSHIK27** [comment]
> @JOSHIK27 once the PR is ready for review, drop a comment

When i ran cypress tests by running docker container locally with loaded dummy data on develop branch(as per readme of backend), bunch of tests failed which shouldn't happen. @nihal467 Am i missing something ?
image

**Comment by nihal467** [comment]
@JOSHIK27 

image

- [ ] After creating a new consultation with medicine for a patient, there is something went wrong error thrown on the frontend

image

- [ ] when you open any live consultation page, it is also throwing something went wrong error

image

- Click on the details of the failing action and try to run the failing test manually; then you can debug it properly. For now, in general, the Cypress failure is due to a bug in the PR.

**Comment by JOSHIK27** [comment]
> @JOSHIK27
> 
> image
> 
> * [ ]  After creating a new consultation with medicine for a patient, there is something went wrong error thrown on the frontend
> 
> image
> 
> * [ ]  when you open any live consultation page, it is also throwing something went wrong error
> 
> image
> 
> * Click on the details of the failing action and try to run the failing test manually; then you can debug it properly. For now, in general, the Cypress failure is due to a bug in the PR.

Sure... Iam working on it.

**Comment by JOSHIK27** [comment]
I ran a test that failed here (facility/homepage),weirdly it passed when locally tested on same branch as the pr.
@nihal467 @rithviknishad 
The other test that failed is patientlogupdate, when i tested locally it failed (1 pass, 4 fail as same as here). I am assuming facility/homepage failed due to timeouts here. I will look into patientlogupdate test manually and figure out.
image
image

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**